app/models/concerns/client_search.rb

```diff
@@ -0,0 +1,77 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module ClientSearch
+  extend ActiveSupport::Concern
+  included do
+    # Requires a block!
+    def self.text_searcher(text)
+      return none unless text.present?
+
+      text.strip!
+      sa = arel_table
+      alpha_numeric = /[[[:alnum:]]-]+/.match(text).try(:[], 0) == text
+      numeric = /[\d-]+/.match(text).try(:[], 0) == text
+      date = /\d\d\/\d\d\/\d\d\d\d/.match(text).try(:[], 0) == text
+      social = /\d\d\d-\d\d-\d\d\d\d/.match(text).try(:[], 0) == text
+      # Explicitly search for only last, first if there's a comma in the search
+      if text.include?(',')
+        last, first = text.split(',').map(&:strip)
+        where = sa[:LastName].lower.matches("#{last.downcase}%") if last.present?
+        if last.present? && first.present?
+          where = where.and(sa[:FirstName].lower.matches("#{first.downcase}%"))
+        elsif first.present?
+          where = sa[:FirstName].lower.matches("#{first.downcase}%")
+        end
+        # Explicity search for "first last"
+      elsif text.include?(' ')
+        first, last = text.split(' ').map(&:strip)
+        where = sa[:FirstName].lower.matches("#{first.downcase}%").
+          and(sa[:LastName].lower.matches("#{last.downcase}%"))
+        # Explicitly search for a PersonalID
+      elsif alpha_numeric && (text.size == 32 || text.size == 36)
+        where = sa[:PersonalID].matches(text.gsub('-', ''))
```

Comment:
@eanders is the PersonalID _always_ 32 or 36 characters for all our clients? These are imported from different HMIS's so they can have different formats.

Maybe we should remove the `text.size` check altogether, because otherwise we would need to remember to update this search to accommodate importing clients from HMIS's with differently formatted Personal IDs!

Reply:
Nevermind, I see we do that below on line 43. So this one is just for UUIDs to remove the dashes.